Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Checkout.h

```diff
@@ -220,6 +220,13 @@ typedef void (^BUYDataGiftCardBlock)(BUYGiftCard * _Nullable giftCard, NSError *
  */
 - (NSOperation *)removeGiftCard:(BUYGiftCard *)giftCard fromCheckout:(BUYCheckout *)checkout completion:(BUYDataCheckoutBlock)block;
 
+/**
+ *	Removes the discount from the checkout.
+ *
+ *  @param checkout An existing BUYCheckout on Shopify
+ */
+- (void)removeDiscountFromCheckout:(BUYCheckout *)checkout;
```

Comment:
@EishanVijay let's mimic the Android approach. It's not 100% foolproof, but it's reasonable. Especially since `code` is the only property of a discount that can be set (`amount` and `applicable` are read-only).